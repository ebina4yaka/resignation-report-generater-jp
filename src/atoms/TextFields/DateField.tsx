import { ChangeEvent, ReactElement } from 'react'
import TextField from '@material-ui/core/TextField'
import useStyles from './useStyles'
import { TextFieldProps } from './TextFieldProps'

export default function DateField(props: TextFieldProps<string>): ReactElement {
  const { value, minValue, setValue, id, label, isRequired } = props
  const classes = useStyles()

  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
    const { target } = event
    setValue(target.value)
  }

  return (
    <TextField
      required={isRequired}
      id={id}
      label={label}
      type="date"
      value={value}
      className={classes.textField}
      InputLabelProps={{
        shrink: true,
      }}
      inputProps={{ min: minValue }}
      autoComplete="off"
      onChange={handleChange}
    />
  )
}
