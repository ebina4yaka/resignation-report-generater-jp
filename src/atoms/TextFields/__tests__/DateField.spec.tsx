import renderer from 'react-test-renderer'
import { act, renderHook } from '@testing-library/react-hooks'
import { fireEvent, queryByAttribute, render } from '@testing-library/react'
import { useState } from 'react'
import DateField from '../DateField'

describe('DateField Tests', () => {
  test('Snapshot test', () => {
    const tree = renderer
      .create(
        <DateField
          id="testDate"
          isRequired
          value="2020-01-01"
          label="テスト"
          setValue={() => {}}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Input onChange Handle Test', () => {
    const { result } = renderHook(() => {
      const [value, setValue] = useState('2020-01-01')
      return { value, setValue }
    })
    const id = 'testDate'
    const dom = render(
      <DateField
        id={id}
        isRequired
        value={result.current.value}
        label="テスト"
        setValue={result.current.setValue}
      />
    )
    const getById = queryByAttribute.bind(null, 'id')

    const inputElement = getById(dom.container, id) as HTMLInputElement

    act(() => {
      fireEvent.change(inputElement, { target: { value: '2020-01-20' } })
    })

    expect(result.current.value).toBe('2020-01-20')
  })
})
