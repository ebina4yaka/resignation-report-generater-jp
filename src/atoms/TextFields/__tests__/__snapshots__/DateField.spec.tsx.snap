// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateField Tests Snapshot test 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-textField-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled Mui-required Mui-required"
    data-shrink={true}
    htmlFor="testDate"
    id="testDate-label"
  >
    テスト
    <span
      aria-hidden={true}
      className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
    >
       
      *
    </span>
  </label>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoComplete="off"
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      disabled={false}
      id="testDate"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="date"
      value="2020-01-01"
    />
  </div>
</div>
`;
